src/test/java/com/example/SpringMongoProject/Controller/StudentController_getStudents_23978b7130_Test.java

```diff
@@ -0,0 +1,52 @@
+// Test generated by RoostGPT for test springMongoDB using AI Type Open AI and AI Model gpt-4
+
+package com.example.SpringMongoProject.Controller;
+
+import com.example.SpringMongoProject.Entity.Student;
+import com.example.SpringMongoProject.Service.StudentServices;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class StudentController_getStudents_23978b7130_Test {
+
+    @InjectMocks
+    private StudentController studentController;
+
+    @Mock
+    private StudentServices studentServices;
+
+    @Test
+    public void testGetStudents_success() {
+        Student student1 = new Student();
+        student1.set_id("1");
+        student1.setStudentname("John");
+
+        Student student2 = new Student();
+        student2.set_id("2");
+        student2.setStudentname("Jane");
+
+        when(studentServices.listAll()).thenReturn(Arrays.asList(student1, student2));
```

Comment:
The error is indicating that when is not applicable for methods with a void return type. Since saveorUpdate is likely a void method, you should use doNothing() to mock its behavior.